src/components/App.js

```diff
@@ -1,15 +1,167 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { BrowserRouter, Route } from 'react-router-dom';
+import { getWeatherInformation, toCelcius, toFahrenheit } from '../utils/weather';
+import getLocation from '../utils/location';
+
+import UVIndex from './UVIndex';
+import Forecast from './Forecast';
+import Nav from './Nav';
+import UnitToggle from './UnitToggle';
 import Weather from './Weather';
 
-const App = () => (
-  <div>
-    <h1>React Boilerplate with Sass</h1>
-    <img
-      src="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg"
-      alt="react-logo"
-    />
-    <Weather />
-  </div>
-);
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      location: false,
+      latitude: null,
+      longitude: null,
+      weatherData: {},
+      forecastData: {},
+      forecastResponse: false,
+      weatherResponse: false,
+      uvResponse: false,
+      uvData: null,
+      temperature: {},
+    };
+
+    this.changeToCelcius = this.changeToCelcius.bind(this);
+    this.changeToFahrenheit = this.changeToFahrenheit.bind(this);
+  }
+
+  componentDidMount() {
+    if ('geolocation' in navigator) {
+      getLocation().then(response => {
+        this.setState({
+          location: true,
+          latitude: response.coords.latitude,
+          longitude: response.coords.longitude,
+        });
+        this.getCurrentWeather();
+        this.getCurrentForecast();
+        this.getUVIndex();
+      });
+    }
+  }
+
+  getCurrentWeather() {
+    const { latitude, longitude, location } = this.state;
+    if (location) {
+      getWeatherInformation('weather', latitude, longitude).then(response => {
+        const { data } = response;
+        this.setState({
+          weatherResponse: true,
+          temperature: { type: 'FAHRENHEIT', value: data.main.temp, unit: 'F' },
+          weatherData: {
+            name: data.name,
+            description: data.weather[0].description,
+            icon: data.weather[0].icon,
+          },
+        });
+      });
+    }
+  }
+
+  getCurrentForecast() {
+    const { latitude, longitude, location } = this.state;
+    if (location) {
+      getWeatherInformation('forecast/daily', latitude, longitude).then(response => {
+        const { data } = response;
+        this.setState({
+          forecastResponse: true,
+          forecastData: data.list,
+        });
+      });
+    }
+  }
+
+  getUVIndex() {
+    const { latitude, longitude, location } = this.state;
+    if (location) {
+      getWeatherInformation('uvi', latitude, longitude).then(response => {
+        const { data } = response;
+        this.setState({
+          uvData: data,
+          uvResponse: true,
+        });
+      });
+    }
+  }
+
+  changeToFahrenheit() {
```

Comment:
this two functions could be joined into one that takes the fahrenheit/Celcius as the parameter

Reply:
done